src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -77,12 +77,7 @@ def gethistory(key, range, cache = 1)
     cachekey = key + "-#{range}"
     Rails.cache.delete(cachekey, :shared => true) if !cache
     return Rails.cache.fetch(cachekey, :expires_in => (range.to_i * 3600) / 150, :shared => true) do
-      hash = Hash.new
-      data = ActiveXML::api.direct_http(URI('/status/history?key=%s&hours=%d&samples=400' % [key, range]))
-      doc = Nokogiri::XML(data)
-      doc.root.elements.each do |value|
-        hash[value.attributes['time'].value.to_i] = value.attributes['value'].value.to_f
-      end
+      hash = StatusHistory.history_by_key_and_hours(key, range)
       hash.sort { |a, b| a[0] <=> b[0] }
```

Comment:
This would mean that the XML coming back from the API changes. I would like to avoid that, who knows what people do with it...